sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1055,96 @@ def _eval_is_nonnegative(self):
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k1 objects in the first bin, k2 objects in the second bin, and so on.
+
+    To expand the multinomial when `k` contains a symbol, use either
+    ``expand_func()`` or ``expand(func=True)``. The former will keep
+    the polynomial in factored form while the latter will expand the
+    polynomial itself. See examples for details.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    >>> multinomial(n, -1)
+    0
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2,1,0)
+    3
+
+    >>> multinomial(Rational(5, 4), 3)
+    gamma(21/4)/(6*gamma(9/4))
+
+    >>> multinomial(n, 3)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(self, *args):
+        args = list(map(sympify, args))
+        if all(x.is_Integer for x in args):
+            from sympy import binomial
+            return Mul(*[binomial(j, k) for j, k in zip([Add(*args[:i]) for i in range(1, len(args) + 1)], args)])
+        elif all(x.is_number for x in args):
+            from sympy import gamma
+            return gamma(Add(*args) + 1)/Mul(*(gamma(x + 1) for x in args))
+
+    def _eval_rewrite_as_factorial(self, *args, **kwargs):
+        return factorial(Add(*args))/Mul(*(factorial(x) for x in args))
+
+    def _eval_expand_func(self, **hints):
+        #return Mul(*[binomial._eval_expand_func(binomial(j, k)) for j, k in zip([Add(*self.args[:i]) for i in range(1, len(self.args) + 1)], self.args)])
+        from sympy import gamma, expand_func
+        return expand_func(gamma(Add(*self.args) + 1)/Mul(*(gamma(x + 1) for x in self.args)))
+
+    def _eval_rewrite_as_binomial(self, *args, **kwargs):
+        from sympy import binomial
+        return Mul(*[binomial(j, k) for j, k in zip([Add(*args[:i]) for i in range(1, len(args) + 1)], args)])
+
+    def _eval_rewrite_as_gamma(self, *k, **kwargs):
+        from sympy import gamma
+        return gamma(Add(*k) + 1)/Mul(*(gamma(x + 1) for x in k))
+
+    def _eval_is_integer(self):
+        k = list(map(sympify, self.args))
+        print(list(x.is_integer for x in k))
+        if all(x.is_integer for x in k):
```

Comment:
`or any(x.is_negative for x in k):`

Reply:
why this tho ? won't is_integer recognise both -ve and +ve's ?

Reply:
To avoid the conflicting definition of negatives in binomial (if that's the problem) I suggest:
```python
if any(x.is_negative for x in k):
    return S.Zero  # non-existing "multinomial"
elif all(x.is_integer for x in k):
    ...
else:
   ...
```

Reply:
Thank you! ... I've solved it in the eval function 😄